exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillConstExecutor.java

```diff
@@ -98,11 +98,37 @@
       //      - CHAR, SYMBOL, MULTISET, DISTINCT, STRUCTURED, ROW, OTHER, CURSOR, COLUMN_LIST
       .build();
 
+  private static ImmutableMap<SqlTypeName, TypeProtos.MinorType> CALCITE_TO_DRILL_MAPPING =
+      ImmutableMap.<SqlTypeName, TypeProtos.MinorType> builder()
+          .put(SqlTypeName.INTEGER, TypeProtos.MinorType.INT)
+          .put(SqlTypeName.BIGINT, TypeProtos.MinorType.BIGINT)
+          .put(SqlTypeName.FLOAT, TypeProtos.MinorType.FLOAT4)
+          .put(SqlTypeName.DOUBLE, TypeProtos.MinorType.FLOAT8)
+          .put(SqlTypeName.VARCHAR, TypeProtos.MinorType.VARCHAR)
+          .put(SqlTypeName.BOOLEAN, TypeProtos.MinorType.BIT)
+          .put(SqlTypeName.DATE, TypeProtos.MinorType.DATE)
+          .put(SqlTypeName.TIME, TypeProtos.MinorType.TIME)
+          .put(SqlTypeName.TIMESTAMP, TypeProtos.MinorType.TIMESTAMP)
+          .put(SqlTypeName.VARBINARY, TypeProtos.MinorType.VARBINARY)
+          .put(SqlTypeName.INTERVAL_YEAR_MONTH, TypeProtos.MinorType.INTERVALYEAR)
+          .put(SqlTypeName.INTERVAL_DAY_TIME, TypeProtos.MinorType.INTERVALDAY)
+          //.put(SqlTypeName.MAP, TypeProtos.MinorType.MAP)
```

Comment:
Some of these types are commented out. Can you add/ correct comments explaining why?

Also, some of the comments are incorrect.